src/rendering/RenderUtil.cc

```diff
@@ -2720,6 +2926,99 @@ void RenderUtil::ViewCOM(const Entity &_entity)
   }
 }
 
+/////////////////////////////////////////////////
+void RenderUtil::ViewJoints(const Entity &_entity)
+{
+  std::vector<Entity> jointEntities;
+  if (this->dataPtr->modelToJointEntities.find(_entity) !=
+           this->dataPtr->modelToJointEntities.end())
+  {
+    jointEntities.insert(jointEntities.end(),
+        this->dataPtr->modelToJointEntities[_entity].begin(),
+        this->dataPtr->modelToJointEntities[_entity].end());
+  }
+
+  if (this->dataPtr->modelToModelEntities.find(_entity) !=
+      this->dataPtr->modelToModelEntities.end())
+  {
+    std::stack<Entity> modelStack;
+    modelStack.push(_entity);
+
+    std::vector<Entity> childModels;
+    while (!modelStack.empty())
+    {
+      Entity model = modelStack.top();
+      modelStack.pop();
+
+      jointEntities.insert(jointEntities.end(),
+          this->dataPtr->modelToJointEntities[model].begin(),
+          this->dataPtr->modelToJointEntities[model].end());
+
+      childModels = this->dataPtr->modelToModelEntities[model];
+      for (const auto &childModel : childModels)
+      {
+        modelStack.push(childModel);
+      }
+    }
+  }
+
+  // Toggle joints
+  bool showJoint, showJointInit = false;
+
+  // first loop looks for new joints
+  for (const auto &jointEntity : jointEntities)
+  {
+    if (this->dataPtr->viewingJoints.find(jointEntity) ==
+        this->dataPtr->viewingJoints.end())
+    {
+      this->dataPtr->newJoints.push_back(_entity);
+      showJointInit = showJoint = true;
+    }
+  }
+
+  // second loop toggles joints
+  for (const auto &jointEntity : jointEntities)
+  {
+    if (this->dataPtr->viewingJoints.find(jointEntity) ==
+        this->dataPtr->viewingJoints.end())
+      continue;
+
+    // when viewing multiple joints (e.g. _entity is a model),
+    // boolean for view joints is based on first jointEntity in list
+    if (!showJointInit)
+    {
+      showJoint = !this->dataPtr->viewingJoints[jointEntity];
+      showJointInit = true;
+    }
+
+    rendering::VisualPtr jointVisual =
+        this->dataPtr->sceneManager.VisualById(jointEntity);
+    if (jointVisual == nullptr)
+    {
+      ignerr << "Could not find visual for entity [" << jointEntity
+             << "]" << std::endl;
+      continue;
+    }
+
+    this->dataPtr->viewingJoints[jointEntity] = showJoint;
+    jointVisual->SetVisible(showJoint);
+
+    if (showJoint)
+    {
+      // turn off wireboxes for joint entity
+      if (this->dataPtr->wireBoxes.find(jointEntity)
+            != this->dataPtr->wireBoxes.end())
+      {
+        ignition::rendering::WireBoxPtr wireBox =
+          this->dataPtr->wireBoxes[jointEntity];
```

Comment:
nitpick: store find result to avoid second lookup:

```cpp
      auto wireBoxIt = this->dataPtr->wireBoxes.find(jointEntity);
      if (wireBoxIt != this->dataPtr->wireBoxes.end())
      {
        ignition::rendering::WireBoxPtr wireBox = wireBoxIt->second;
```